ee/clickhouse/views/person.py

```diff
@@ -1,30 +1,94 @@
+from posthog.constants import FunnelVizType
 from posthog.decorators import cached_function
+from posthog.exceptions import UnsupportedFeature
 from posthog.models import Event, Filter, Person
 
 
-class ClickhousePersonViewSet(PersonViewSet):
+def filter_persons_ch(team_id: int, request: Request, queryset: QuerySet) -> QuerySet:
+    # Keep functionality in sync with posthog/api/person.py
+    params: Dict[str, Any] = {"team_id": team_id}
+    and_conditions: List[str] = []
+    if request.GET.get("id"):
+        raise UnsupportedFeature("filtering persons by field `id`")
+    if request.GET.get("uuid"):
+        params["uuids"] = request.GET["uuid"].split(",")
+        and_conditions.append("AND has(%(uuids)s, toString(person.id))")
+    if request.GET.get("search"):
+        parts = request.GET["search"].split(" ")
+        contains = []
+        for part_index, part in enumerate(parts):
+            if ":" in part:
+                matcher, key = part.split(":")
+                if matcher == "has":
+                    # Matches for example has:email or has:name
+                    params[f"has_key_{part_index}"] = key
+                    and_conditions.append(f"AND JSONHas(properties, %(has_key_{part_index})s)")
+            else:
+                contains.append(part)
+        if contains:
+            params["icontains"] = f'%{" ".join(contains)}%'
+            and_conditions.append(
+                f"""
+                AND (
+                    properties ILIKE %(icontains)s
+                    OR arrayExists(distinct_id -> ilike(distinct_id, %(icontains)s), distinct_ids)
+                )"""
+            )
+    properties: List[Dict[str, Any]] = json.loads(request.GET["properties"]) if request.GET.get("properties") else []
+    if request.GET.get("cohort"):
+        properties.append({"type": "cohort", "key": "id", "value": request.GET["cohort"]})
+    if properties:
+        for property in properties:
+            if property.get("type") is None:
+                # In this endpoint the default type is "person", not default "event", which we ensure here
+                property["type"] = "person"
+        filter = Filter(data={"properties": properties})
+        # Materialized columns are disabled as they don't work with GET_LATEST_PERSON_WITH_DISTINCT_IDS_SQL's any()s
+        filter_query, filter_params = parse_prop_clauses(
+            filter.properties, team_id, is_person_query=True, allow_denormalized_props=False
+        )
+        and_conditions.append(filter_query)
+        params.update(filter_params)
+
+    if and_conditions:
+        final_query = GET_LATEST_PERSON_WITH_DISTINCT_IDS_SQL.format(
+            query=" ".join(and_conditions).replace("AND ", "", 1)
+        )
+        uuids_found_rows = cast(list, sync_execute(final_query, params))
+        uuids_found = [row[0] for row in uuids_found_rows]
+        queryset = queryset.filter(uuid__in=uuids_found)
```

Comment:
This smells like a hack.

Why not use `ClickhousePersonSerializer` like other endpoints? Going to postgres in the end is kind of fishy without a good cause.